### ISSUE TITLE
Unable to translate checkout country names

### ISSUE THREAD

**Issue by rokslavec77** [magento/magento2#22956]
### Summary (*)

**Magento 2.3-develop**
I want to be able to change the translations for the country options names that display on the checkout shipping and billing address forms. Usual translations through csv don't work.

I have found that the current translations are located in this folder: /vendor/magento/zendframework1/library/Zend/Locale/Data/
I have tried to edit the files inside (even thought its vendor), but it's still not working.

### Steps to reproduce (*)

1. Setup website, store and store view with locale mk_MK.
2. Added translations file: app/design/frontend/Optiweb/owtheme/i18n/mk_MK.csv with content:
```
"Country","Country Translated"
"Северна Македонија","severna makedonija"
```
3. Flushed cache.
4. Add product to cart and go to checkout shipping address form

### Expected result (*)
All will be translated
### Actual result (*)
Translation works only for "Country"
![actual2](https://user-images.githubusercontent.com/51680850/62693243-cf185c00-b9da-11e9-9ca5-b4e24a6fa5de.png)

### Proposed solution

I think there should be a way to change this translations (csv, from admin or any other option). In our example the country on checkout is the only string that is in Cryrilic script, which is bad user experience. And also the country name changed this year to "Northern Macedonia", so we should be able to change it somehow.

**Comment by rokslavec77** [comment]
<h3>Summary:</h3>
I want to be able to change the translations for the country options names that display on the checkout shipping and billing address forms. Usual translations through csv don't work.

I have found that the current translations are located in this folder: /vendor/magento/zendframework1/library/Zend/Locale/Data/
I have tried to edit the files inside (even thought its vendor), but it's still not working.

How can that translation be changed?

<h3>Information on your environment:</h3>

Magento CE 2.3.1 with sample data is installed.

<h3>Steps to reproduce:</h3>

1. Setup website, store and store view with locale mk_MK.

2. Added translations file:
app/design/frontend/Optiweb/owtheme/i18n/mk_MK.csv
with content:
"Country","Country Translated"
"Македонија","severna makedonija"

3. Flushed cache.

4. Add product to cart and go to checkout shipping address form

<h3>Expected result:</h3>

Both translations should work.

<h3>Actual result:</h3>

The string translation works, country option is not translated

![image](https://user-images.githubusercontent.com/50859584/58074757-4d1c5200-7ba6-11e9-8147-3b64f463e15f.png)

**Comment by rokslavec77** [comment]
Reproduced on 2.3.x
Reproduced on 2.2.x

**Comment by engcom-Charlie** [comment]
Hello @rokslavec77 
I can't reproduce this issue on fresh Magento 2.3-develop.
**Manual testing scenario:**
1. Set Slovenian locale
2. Go to the Checkout page
3. Open Country dropdown
4. Check translation
**Actual result:**
All countries translated:
![image](https://user-images.githubusercontent.com/51680850/62689131-e2bfc480-b9d2-11e9-8ff5-f72fa521b09c.png)
So i close this issue.

**Comment by rokslavec77** [comment]
You tested it with Slovenian locale. The issue happens with locales that use cyrilic (for example Macedonian):

Steps to reproduce:

1. Setup website, store and store view with locale mk_MK.

2. Added translations file:
app/design/frontend/Optiweb/owtheme/i18n/mk_MK.csv
with content:
"Country","Country Translated"
"Македонија","severna makedonija"

3. Flushed cache.

4. Add product to cart and go to checkout shipping address form

**Comment by engcom-Charlie** [comment]
@rokslavec77 thank you for contributing!
Result:
Translated only "Country" for "Country Translated":
![actual2](https://user-images.githubusercontent.com/51680850/62692972-46012500-b9da-11e9-9c37-67f2ec09bacb.png)
I will confirm your issue but i have to update description

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19029` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @rokslavec77. Thank you for your report. 
The issue has been fixed in magento/magento2#24179 by @Bartlomiejsz in 2.3-develop branch
Related commit(s):
 - [b423aa08f7459822b33e2b55d18e6e21788a4c33](https://github.com/magento/magento2/commit/b423aa08f7459822b33e2b55d18e6e21788a4c33)
 - [23afffb2d62b63fce11c80f04f864e89ff748561](https://github.com/magento/magento2/commit/23afffb2d62b63fce11c80f04f864e89ff748561)
 - [8dab512ee45349f84f6ac71387199b09301c5c75](https://github.com/magento/magento2/commit/8dab512ee45349f84f6ac71387199b09301c5c75)

The fix will be available with the upcoming 2.3.4 release.
